gen/zap_test.go

```diff
@@ -852,3 +853,26 @@ func TestLogNilStruct(t *testing.T) {
 	require.NoError(t, x.MarshalLogObject(enc))
 	assert.Empty(t, enc.Fields)
 }
+
+func TestZapOptOut(t *testing.T) {
```

Comment:
Is this a new test? Should we also be adding a the case here for a redacted field since we're adding this new feature here?

Reply:
Yes, it is. I noticed that no tests have been conducted for ZapOut. Implementing a small test could help increase our code coverage. I updated the test and add a test case.